crates/noirc_frontend/src/monomorphization/mod.rs

```diff
@@ -652,6 +652,10 @@ impl<'interner> Monomorphizer<'interner> {
                 ast::Type::Tuple(fields)
             }
 
+            HirType::Trait(def, _args) => {
+                unreachable!("Not sure what to do here def = {:?}", def)
+            }
```

Comment:
There is nothing to do because traits are not types and we cannot convert them into types.